From 868ffc49a3a29b9d09c0bd01d05e872eaabc64c1 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sat, 2 Nov 2024 11:04:47 +0100
Subject: [PATCH] Allow explicit setting of Python3 fallback executible path

This allows correct configuration of this path for cross-building
setups where the python3 executible lives in a sysroot path and only the
subpath inside the sysroot shall be used.

Upstream-Status: Submitted [https://invent.kde.org/frameworks/ki18n/-/merge_requests/132]

SPDX-FileCopyrightText: Andreas Cord-Landwehr <cordlandwehr@kde.org>
SPDX-License-Identifier: MIT
Signed-off-by: Andreas Cord-Landwehr <cordlandwehr@kde.org>
---
 CMakeLists.txt               | 4 ++++
 cmake/KF6I18nMacros.cmake.in | 4 +---
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cac6ca1..7203a76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,10 @@ set_package_properties(IsoCodes PROPERTIES TYPE RUNTIME
 # we have to hardcode the PYTHON_EXECUTABLE path or anything depending on KF6I18n
 # would be unable to find another Python version.
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
+# allow settings this path explicitly for cross-building setups
+if (NOT FALLBACK_KI18N_PYTHON_EXECUTABLE)
+    set(FALLBACK_KI18N_PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
+endif()
 configure_file(cmake/KF6I18nMacros.cmake.in ${KI18n_BINARY_DIR}/cmake/KF6I18nMacros.cmake @ONLY)
 # Needed to build the ki18n translations and run the autotests
 file(COPY ${KI18n_SOURCE_DIR}/cmake/build-pofiles.cmake DESTINATION ${KI18n_BINARY_DIR}/cmake)
diff --git a/cmake/KF6I18nMacros.cmake.in b/cmake/KF6I18nMacros.cmake.in
index 07b9be1..d9c7c96 100644
--- a/cmake/KF6I18nMacros.cmake.in
+++ b/cmake/KF6I18nMacros.cmake.in
@@ -6,11 +6,9 @@ find_package(Gettext REQUIRED)
 
 # The Python executable used for building ki18n will be used as a fallback
 # solution if it cannot be found in $PATH when building applications.
-set(_KI18N_PYTHON_EXECUTABLE "@Python3_EXECUTABLE@")
-
 find_program(KI18N_PYTHON_EXECUTABLE NAMES python3 python2 python)
 if(NOT KI18N_PYTHON_EXECUTABLE)
-    set(KI18N_PYTHON_EXECUTABLE "${_KI18N_PYTHON_EXECUTABLE}")
+    set(KI18N_PYTHON_EXECUTABLE "@FALLBACK_KI18N_PYTHON_EXECUTABLE@")
 endif()
 
 set(_ki18n_pmap_compile_script ${CMAKE_CURRENT_LIST_DIR}/ts-pmap-compile.py)
-- 
2.43.0

